### DISCUSSION TITLE
Strategies for building large amount of pages

### DISCUSSION THREAD

**Discussion by M1ck0** [vercel/next.js#18166] [Help] [answered]
What is the best way to build a big amount of pages?

This is WordPress headless CMS that has 90k pages. I am using GraphQL to fetch data.

Is there a way to optimize this because now it takes a lot of time to build them?

Is it good for all pages to be static with the revalidate option to update content?

Are there any drawbacks to using this method, and is it possible to overcome them? If not, what do you suggest?

I am pretty flexible and I can easily change the flow of the project for now.

**Comment by jamesmosier** [comment, marked as answer]
Hi @M1ck0. The recommended way to handle a huge amount of pages is to use the [`fallback: true`](https://nextjs.org/docs/basic-features/data-fetching#fallback-true) option in `getStaticProps`. Using this along with `revalidate` is a great option for a large amount of pages.
